src/routes/safe/components/Transactions/TxsTable/ExpandedTx/TxDescription/CustomDescription.tsx

```diff
@@ -0,0 +1,183 @@
+import EtherscanLink from 'src/components/EtherscanLink'
+import { humanReadableValue } from 'src/utils/humanReadableValue'
+import Collapse from 'src/components/Collapse'
+import { useSelector } from 'react-redux'
+import { getNameFromAddressBook } from 'src/logic/addressBook/store/selectors'
+import Paragraph from 'src/components/layout/Paragraph'
+import LinkWithRef from 'src/components/layout/Link'
+import { shortVersionOf } from 'src/logic/wallets/ethAddresses'
+import { Transaction } from 'src/routes/safe/store/models/types/transaction'
+
+export const TRANSACTIONS_DESC_CUSTOM_VALUE_TEST_ID = 'tx-description-custom-value'
+export const TRANSACTIONS_DESC_CUSTOM_DATA_TEST_ID = 'tx-description-custom-data'
+
+const useStyles = makeStyles(styles)
+
+const TxDetailsMethodName = styled(Text)`
+  text-indent: 4px;
+`
+const TxDetailsMethodParam = styled.div`
+  text-indent: 8px;
+`
+const InlineText = styled(Text)`
+  display: inline-flex;
+`
+const TxDetailsContent = styled.div`
+  padding: 8px 8px 8px 16px;
+`
+
+const TxInfo = styled.div`
+  padding: 8px 8px 8px 16px;
+`
+
+const MultiSigCustomData = ({ tx, order }: { tx: MultiSendDetails; order: number }): React.ReactElement => {
+  const classes = useStyles()
+  const methodName = tx.data?.method ? ` (${tx.data.method})` : ''
+
+  return (
+    <>
+      <Collapse
+        collapseClassName={classes.collapse}
+        headerWrapperClassName={classes.collapseHeaderWrapper}
+        title={<IconText iconSize="sm" iconType="code" text={`Action ${order + 1}${methodName}`} textSize="lg" />}
+      >
+        <TxDetailsContent>
+          <TxInfo>
+            <Bold>Send {humanReadableValue(tx.value)} ETH to:</Bold>
+            <OwnerAddressTableCell address={tx.to} showLinks />
+          </TxInfo>
+          {tx.data && (
+            <TxInfo>
+              <TxDetailsMethodName size="lg">
+                <strong>{tx.data.method}</strong>
+              </TxDetailsMethodName>
+              {tx.data?.parameters.map((param, index) => (
+                <TxDetailsMethodParam key={`${tx.operation}_${tx.to}_${tx.data.method}_param-${index}`}>
+                  <InlineText size="lg">
+                    <strong>
+                      {param.name}({param.type}):
+                    </strong>
+                  </InlineText>
+                  <RenderValue method={methodName} type={param.type} value={param.value} />
+                </TxDetailsMethodParam>
+              ))}
+            </TxInfo>
+          )}
+        </TxDetailsContent>
+      </Collapse>
+    </>
+  )
+}
+
+const TxData = ({ data }: { data: string }): React.ReactElement => {
+  const classes = useStyles()
+  const [showTxData, setShowTxData] = React.useState(false)
+  const showExpandBtn = data.length > 20
+
+  return (
+    <Paragraph className={classes.txDataParagraph} noMargin size="md">
+      {showExpandBtn ? (
+        <>
+          {showTxData ? (
+            <>
+              {data}{' '}
+              <LinkWithRef
```

Comment:
can we move the expand button somewhere else?  in case if expanded data is super long, user would have to scroll down for "show less" button

Reply:
but what if the user scrolled down to the end of the expanded data?

Anyway, this should be discussed in a different issue and with some discussion about UX